File: app/src/main/java/com/company/chap22mathgame/MainActivity.java
package com.company.chap22mathgame;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button add, sub, mul;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        add = findViewById(R.id.add);
        sub = findViewById(R.id.sub);
        mul = findViewById(R.id.mul);

        add.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, GameplayActivity.class);
                intent.putExtra("mode", "addition");
                startActivity(intent);
                finish();
            }
        });
        sub.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, GameplayActivity.class);
                intent.putExtra("mode", "subtraction");
                startActivity(intent);
                finish();
            }
        });
        mul.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, GameplayActivity.class);
                intent.putExtra("mode", "multiplication");
                startActivity(intent);
                finish();
            }
        });
    }
}

File: app/src/main/java/com/example/chap11mathgame/MainActivity.kt
package com.example.chap11mathgame

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {

    private lateinit var add: Button
    private lateinit var sub: Button
    private lateinit var mul: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        add = findViewById(R.id.add)
        sub = findViewById(R.id.sub)
        mul = findViewById(R.id.mul)

        add.setOnClickListener {
            var intent = Intent(this@MainActivity, GameplayActivity::class.java)
            intent.putExtra("mode", "addition")
            startActivity(intent)
        }
        sub.setOnClickListener {
            var intent = Intent(this@MainActivity, GameplayActivity::class.java)
            intent.putExtra("mode", "subtraction")
            startActivity(intent)
        }
        mul.setOnClickListener {
            var intent = Intent(this@MainActivity, GameplayActivity::class.java)
            intent.putExtra("mode", "multiplication")
            startActivity(intent)
        }

    }
}